lib/Doctrine/DBAL/Driver/Ibase/AbstractIbaseStatement.php

```diff
@@ -0,0 +1,672 @@
+
+    /**
+     * Fetch all records into an array of numeric arrays
+     * @return type
+     */
+    protected function internalFetchAllNum()
+    {
+        $result = array();
+        while ($data = $this->internalFetchNum()) {
+            $result[] = $data;
+        }
+        return $result;
+    }
+
+    /**
+     * Fetches all records into an array containing arrays with column name and column index 
+     * @return array
+     */
+    protected function internalFetchAllBoth()
+    {
+        $result = array();
+        while ($data = $this->internalFetchBoth()) {
+            $result[] = $data;
+        }
+        return $result;
+    }
+
+    /**
+     * Fetches into an object
+     * 
+     * @param object|string $aClassOrObject Object instance or class
+     * @param array $constructorArguments   Parameters to pass to the constructor
+     * @return object|false                 Result object or false
+     */
+    protected function internalFetchClassOrObject($aClassOrObject, array $constructorArguments = null)
+    {
+        $rowData = $this->internalFetchAssoc();
+        if (is_array($rowData)) {
+            return $this->createObjectAndSetPropertiesCaseInsenstive($aClassOrObject, is_array($constructorArguments) ? $constructorArguments : array(), $rowData);
+        } else {
+            return $rowData;
+        }
+    }
+
+    /**
+     * Fetches all records into objects
+     * 
+     * @param object|string $aClassOrObject Object instance or class
+     * @param array $constructorArguments   Parameters to pass to the constructor
+     * @return object|false                 Result object or false
+     */
+    protected function internalFetchAllClassOrObjects($aClassOrObject, array $constructorArguments)
+    {
+        $result = array();
+        while ($row = $this->internalFetchClassOrObject($aClassOrObject, $constructorArguments)) {
+            if ($row !== false) {
+                $result[] = $row;
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Fetches the next record into an associative array
+     * @return array
+     */
+    protected function internalFetchAssoc()
+    {
+        return @ibase_fetch_assoc($this->ibaseResultRc, IBASE_TEXT);
+    }
+
+    /**
+     * Fetches the next record into an array using column name and index as key
+     * @return array|boolean
+     */
+    protected function internalFetchBoth()
+    {
+        $tmpData = ibase_fetch_assoc($this->ibaseResultRc, IBASE_TEXT);
+        if (!$tmpData === FALSE)
+            return array_merge(array_values($tmpData), $tmpData);
+        else
+            return FALSE;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function bindValue($param, $value, $type = null)
+    {
+        return $this->bindParam($param, $value, $type, null);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function bindParam($column, &$variable, $type = null, $length = null)
+    {
+        if (is_array($variable))
```

Comment:
The ibase-api itself does not offer named parameters, too, but I simulate them using an internal name => position map because they are much more convinient. And real pepared statements are better for the firebird server. 

But if none of the other drivers support lists it might be better to remove support for them in the firebird-driver, too, as the support of lists adds complexity and causes a special case where a naive prepare cannot be used anyway because of the parameter-dependant necessary query-rewrite.